tensorflow/core/platform/cloud/retrying_file_system.cc

```diff
@@ -37,55 +42,81 @@ bool IsRetriable(Status status) {
   }
 }
 
-Status CallWithRetries(const std::function<Status()>& f) {
-  int attempts = 0;
+void WaitBeforeRetry(const int delay_seconds) {
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_int_distribution<> dist(0, 999);
+
+  const int delay_milliseconds = delay_seconds * 1000;
+  const int random_milliseconds = dist(gen);
+  const int delay = std::min(delay_milliseconds + random_milliseconds,
+                             kMaximumBackoffMilliSeconds);
+
+  std::this_thread::sleep_for(std::chrono::milliseconds(delay));
+}
+
+Status CallWithRetries(const std::function<Status()>& f,
+                       const int initial_delay_seconds) {
+  int retries = 0;
   while (true) {
-    attempts++;
     auto status = f();
-    if (!IsRetriable(status) || attempts >= kMaxAttempts) {
+    if (!IsRetriable(status) || retries >= kMaxRetries) {
       return status;
     }
     LOG(ERROR) << "The operation resulted in an error and will be retried: "
```

Comment:
I wonder if you could also print the delay in seconds in the LOG message, e.g 
const int delay_seconds = initial_delay_seconds << retries;
... will be retried after " << delay_seconds << " sec:"